crates/bevy_ecs/src/event.rs

```diff
@@ -337,9 +337,9 @@ impl<E: Event> Default for ManualEventReader<E> {
 #[allow(clippy::len_without_is_empty)] // Check fails since the is_empty implementation has a signature other than `(&self) -> bool`
 impl<E: Event> ManualEventReader<E> {
     /// See [`EventReader::iter`]
-    pub fn iter<'a>(
+    pub fn iter<'a, 'b: 'a>(
         &'a mut self,
-        events: &'a Events<E>,
+        events: &'b Events<E>,
```

Comment:
I think mostly because of a test, but I'll check a bit later

Reply:
> I think this might be better anyways since it allows events to not be the exact same lifetime as the reader

That's not quite true. When resolving a lifetime for `'a` in the original code, rustc will choose whichever of the two lifetimes is smaller. They don't need to be exact, you just need one of them to outlive the other. The updated code requires that the event queue outlive the event reader, which is basically the same constraint as the original but slightly more restrictive.

Reply:
ya actually I'm not sure why I changed this, it seems to work fine.